src/main/java/com/shift4/Shift4Gateway.java

```diff
@@ -62,10 +62,18 @@ public Charge createCharge(ChargeRequest request) {
         return post(CHARGES_PATH, request, Charge.class);
     }
 
+    public Charge createCharge(ChargeRequest request, RequestOptions requestOptions) {
+        return post(CHARGES_PATH, request, requestOptions, Charge.class);
+    }
```

Comment:
We should add idempotency key to ChargeRequest as json ignored and use it in client to add header to not multiply methods

Reply:
RequestOptions is quite a standard way of doing this kind of feature. I wouldn't say I like the idea, of messing up the request body with some additional data.

Reply:
Ok, lets leave this as separate param then